sympy/ntheory/ecm.py

```diff
@@ -0,0 +1,221 @@
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                   Lenstra's Elliptic Curve Factorization                   #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class Point:
+    """For denoting the point on the curve.
+    For ECM we don't need the y-coordinate. All
+    arithmetic operations use only the x and z coordinates.
+    """
+
+    def __init__(self, x, z):
+        self.x = x
+        self.z = z
+
+def _pt_add(P, Q, diff, n):
+    """
+    Add two points P and Q where diff = P - Q.
+
+    Parameters:
+    ===========
+
+    P : point on the curve in Montgomery form
+    Q : point on the curve in Montgomery form
+    diff = P - Q
+    n = modulus
+    """
+
+    u = (P.x - P.z)*(Q.x + Q.z)
+    v = (P.x + P.z)*(Q.x - Q.z)
+    s, d = u + v, u - v
+    x = diff.z*s*s % n
```

Comment:
Add in the documentation the mathematics of adding points in the Montogomery form. Also, use variable names more readable for eg. `s` -> `sum`. Also as I don't know much about mathematics of points in Montogomery form. I will try to check the logic after thoroughly understanding about it.

Reply:
The psedocode for Montgomery `addition`, `doubling`, `ladder` is from http://www.hyperelliptic.org/tanja/SHARCS/talks06/Gaj.pdf
And the `ecm`      function pseudocode is available in the book Prime numbers computational perspective. I have added these in references.